src/main/kotlin/org/rust/ide/formatter/impl/spacing.kt

```diff
@@ -254,8 +259,10 @@ private fun SpacingContext.blockMustBeMultiLine(): Boolean {
     }
 }
 
+/**
+ * Assumes that left is before right and they are siblings, otherwise bad things will happen.
+ */
 private fun countNonWhitespaceASTNodesBetween(left: ASTNode, right: ASTNode): Int {
-    // TODO(jajakobyly): assert that left is before right and they are siblings
```

Comment:
Hm, would `check(left.parent == right.parent && left.startOffset < right.startOffset)` do the necessary checks in O(1)?

Kotlin has `assert` which is the same as Javas `assert`. It is disabled or enabled depending on the runtime flag.

There are also design by contract style  `require` (preconditions) and `check` (invariant) functions which are always on. I usually use those.